backend/src/polling_station/structs.rs

```diff
@@ -73,33 +79,126 @@ impl Validate for PollingStationResults {
+                    ],
+                    code: ValidationResultCode::EqualInput,
+                });
+            }
+        // if recounted = false
+        } else {
+            total_voters_counts = self.voters_counts.total_admitted_voters_count;
+            self.voters_counts.validate(
+                election,
+                validation_results,
+                format!("{field_name}.voters_counts"),
+            );
+
+            // W.29 validate that the numbers in voters_counts and votes_counts are not the same
+            if identical_voters_counts_and_votes_counts(&self.voters_counts, &self.votes_counts) {
+                validation_results.warnings.push(ValidationResult {
+                    fields: vec![
+                        format!("{field_name}.voters_counts"),
+                        format!("{field_name}.votes_counts"),
+                    ],
+                    code: ValidationResultCode::EqualInput,
+                });
+            }
+        }
+
+        // F.21 (recounted = false) or F.22 (recounted = true)
+        // validate that the difference for more ballots counted is correct
+        if total_voters_counts < total_votes_counts
+            && (total_votes_counts - total_voters_counts
+                != self.differences_counts.more_ballots_count)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![format!("{field_name}.more_ballots_count")],
+                code: ValidationResultCode::IncorrectDifference,
+            });
+        }
+
+        // F.23 (recounted = false) or F.24 (recounted = true)
+        // validate that the difference for fewer ballots counted is correct
+        if total_voters_counts > total_votes_counts
+            && (total_voters_counts - total_votes_counts
+                != self.differences_counts.fewer_ballots_count)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![format!("{field_name}.fewer_ballots_count")],
+                code: ValidationResultCode::IncorrectDifference,
+            });
+        }
+
+        // F.25 validate that only more or fewer ballots counted is filled in when there is a difference in the totals
+        if total_voters_counts != total_votes_counts
+            && (self.differences_counts.more_ballots_count != 0
+                && self.differences_counts.fewer_ballots_count != 0)
+        {
+            validation_results.errors.push(ValidationResult {
+                fields: vec![
+                    format!("{field_name}.more_ballots_count"),
+                    format!("{field_name}.fewer_ballots_count"),
+                ],
+                code: ValidationResultCode::ConflictingDifferences,
+            });
+        }
+
+        // W.31 (recounted = false) or W.32 (recounted = true)
+        // validate that no difference should be filled in when there is no difference in the totals
```

Comment:
Shouldn't this be an error?

Reply:
No, because it's possible people filled in the paperwork wrong, so it should be possible to fill in what the paper says, but it should give a warning, something is wrong here. This was discussed before with @jorisleker

Reply:
That line of reasoning can be applied to all current errors, leaving us only with warnings.

I thought the distinction between errors and warnings was: errors for what can't be right, warnings for what seems odd but can be right. Filling in a difference when there is none, sounds to me like the former, not the latter.

Reply:
Not sure for this one. This section of the proces-verbaal is pretty confusing, and it does not really guide the user in how these numbers should add up (which is something we can fix of course). Are there any rules / checks performed by CSB on these fields?

![image](https://github.com/user-attachments/assets/1d626d10-d8ef-4b00-a4cb-535ad29a9cbd)

Reply:
The CSB checks for 'onverklaarde verschillen' ('geen verklaring' in this model) in the controleprotocol, but there is no guarantee that these differences add up because OSV2020 doesn't check them.

Reply:
Good points about the proces-verbaal and the CSB. An additional perspective is the current user experience:

1. You fill in "more ballots counted", while you shouldn't have.
2. You get an IncorrectTotal error.
3. You fix the error.
4. You get a NoDifferenceExpected warning.

That's weird, to have the user correct an error to then give them a warning about the actual problem.

Of course, step 3. in my description is very moot. Either you copied the proces-verbaal correctly, or you didn't. That might take several steps and the errors and warnings are there to help you get there. There is no "fixing the error" until the errors/warnings go away. At most at the end there's an issue with the proces-verbaal and we show a warning/error for those.

So it's more of a question which errors/warnings take precedence over others. I think NoDifferenceExpected is the most important one on this page. Which means it should be an error, because we don't show warnings if there are errors.

Reply:
Would it make sense to calculate the "more ballots counted" field in the software based on the difference between the cast votes and admitted voters instead of having the user fill this in? That would solve some of the ways in which the fields can be incorrectly filled.

Reply:
@chrismostert I don't think we're supposed to calculate any of the input fields, because the fields need to be copied from the proces-verbaal into the application. If we'd want to consider this, we should check with our legal team.

Reply:
It also seems odd to me that when there is no difference between voters and votes and you fill both "more" and "fewer", you get a warning, i.e. NoDifferenceExpected. While if there is a difference and you fill both "more" and "fewer", you get an error, i.e. ConflictingDifferences.

Reply:
Decision was made to keep one category of errors: difference between D/D.2 and H should be equal to the more/fewer ballots counted field (F.301 to F.304). All other validations for the differences page result in warnings (W.301 to W.306), not errors.